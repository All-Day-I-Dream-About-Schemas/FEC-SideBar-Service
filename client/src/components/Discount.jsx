import React from 'react';
import styled from 'styled-components';

const DiscountDiv = styled.h5`
  font-family: AdihausDIN,Helvetica,Arial,sans-serif;
  font-weight: 700;
  letter-spacing: 2px;
  text-transform: uppercase;
`;

const Description = styled.span`
  font-family: AdihausDIN,Helvetica,Arial,sans-serif;
  font-size: 14px;
`;

const Discount = ()=> (
  <div>
    <DiscountDiv>Up to 25% off</DiscountDiv>
    <Description>Get 15% off with code SAVINGS at checkout. Spend $120 or more to unlock 25% off your order. Exclusions apply.</Description>
  </div>
)
export default Discount;
