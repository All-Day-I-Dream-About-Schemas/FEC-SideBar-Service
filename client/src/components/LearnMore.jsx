import React from 'react';
import styled from 'styled-components';

const LearnMoreButton = styled.button`
  font-family: AdihausDIN,Helvetica,Arial,sans-serif;
  background-color: transparent;
  border: none;
  color: #000;
  cursor: pointer;
  display: inline;
  padding-left: 1px;
  padding-right: 1px;
  text-decoration: underline;
  &:hover {
    background-color: #000;
    color: #fff;
  }
    `;

  class LearnMore extends React.Component {
    constructor(props) {
      super(props);
      this.handleClick = this.handleClick.bind(this);
    }

    handleClick() {
      this.props.Func();
    }
    render() {
      return(
        <LearnMoreButton onClick={this.handleClick}>Learn more</LearnMoreButton>
      )
    }
  }


  LearnMoreButton.displayName = 'LearnMoreButton';
  export default LearnMore;









// const LearnMoreButton = styled.button`
//   background: none;
//   border: none;
//   color: inherit;
//   font: inherit;
//   font-style: inherit;
//   padding: 0;
//   text-align: inherit;
//   text-transform: inherit;
// `;
