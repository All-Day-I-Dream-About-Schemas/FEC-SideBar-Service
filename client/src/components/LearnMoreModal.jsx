import React, {forwardRef, useImperativeHandle} from 'react';
import styled from 'styled-components';

const Body = styled.div`
  box-sizing: border-box;
  display: flex;
  margin: 0;
  max-height: 92vh;
  padding: 0;
`;

const Wrapper = styled.div`
  bottom: 0;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
`;

const BackDrop = styled.div`
  background-color: rgba(0,0,0,0.6);
  bottom: 0;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 0;
`;

const Box = styled.div`
  border-bottom: 1px solid #000;
  border-left: 1px solid #000;
  border-radius: 0;
  background-color: #fff;
  border-right: 1px solid #000;
  border-top: 1px solid #000;
  height: 22%;
  left: 50%;
  position: relative;
  overflow-y: auto;
  padding-top: 20px;
  padding-bottom: 20px;
  padding-left: 20px;
  padding-right:20px;
  top: 50%;
  transform: translate(-50%, -50%);
  width: 45%;
  z-index: 1;
`;

const Title = styled.h5`
  font-family: AdineuePRO,Helvetica,Arial,sans-serif;
  font-size: 26px;
  font-style: normal;
  font-weight: 600;
  line-height: 24px;
  margin-top: 1px;
  margin-bottom: 18px;
  padding-right: 15px;
  text-transform: uppercase;
`;

const Text = styled.div`
  color: rgba(0,0,0,0.7);
  font-size: 14px;
  font-family: AdihausDIN,Helvetica,Arial,sans-serif;
  font-style: normal;
  font-weight: 400;
  white-space: pre-line;
`;

const CloseButton = styled.button`
  background: #fff;
  border: 1px solid #000;
  color: #000;
  cursor: pointer;
  font-size: 25px;
  height: 40px;
  position: absolute;
  right: 24.5%;
  text-align: center;
  top: 34.5%;
  width: 40px;
`;


const LearnMoreModal = forwardRef((props, ref) => {
  const [display, setDisplay] = React.useState(false);

  useImperativeHandle(ref, () => {
    return {
      open: () => handleOpen(),
      close: () => handleClose()
    }
  });

  const handleOpen = () => {
    setDisplay(true);
  }

  const handleClose = () => {
    setDisplay(false);
  }

  if (display) {
    return(
      <Body>
        <Wrapper>
          <BackDrop onClick={handleClose}>
            <Box>
              <Title>Paying in installments</Title>
              <Text>Want to spread out the cost of your order with adidas? Select ‘Affirm’ or ‘Klarna‘ as your payment method at checkout, and you can pay in interest-free installments, with no hidden fees. Following a quick eligibility check, you’re all set!
              You will be redirected to Affirm's or Klarna´s website.​ </Text><br></br>

              <Text>Affirm payments are subject to eligibility check and approval. Estimated payment amount excludes taxes and shipping fees. Affirm Payment options are provided by Cross River Bank, Member FDIC.​</Text>
            </Box>
          </BackDrop>
        </Wrapper>
        <CloseButton onClick={handleClose}>&Chi;</CloseButton>
      </Body>
    )
  }
  return null
})



export default LearnMoreModal;
