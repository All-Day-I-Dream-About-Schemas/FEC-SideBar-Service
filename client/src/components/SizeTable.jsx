import React from 'react';
import styled from 'styled-components';

const Table = styled.div`
  display: block;
  color: #000;
  font-weight: 400;
  font-style: normal;
  font-family: AdihausDIN,Helvetica,Arial,sans-serif;
  background-color: #fff;
  overflow: auto;
`;

const TableInner = styled.div`
  min-width: 100%;
  width: max-content;
  height: max-content;
`;

const TableRow = styled.div`
  align-items: stretch;
  display: flex;
  width: max-content;
  min-width: 100%;

`;

const TableCell = styled.div`
  align-items: center;
  box-sizing: border-box;
  background-color: #000;
  border-right: 1px solid #000;
  color: #fff;
  display: flex;
  font-size: 14px;
  font-family: AdihausDIN,Helvetica,Arial,sans-serif;
  font-style: normal;
  font-weight: 700;
  left: 0;
  min-height: 50px;
  position: sticky;
  padding: 10px;
  width: 130px;
  // z-index: 1;
`;

const TableCell2 = styled.div`
  align-items: center;
  box-sizing: border-box;
  border-right: 1px solid #000;
  color: #000;
  display: flex;
  font-size: 14px;
  font-family: AdihausDIN,Helvetica,Arial,sans-serif;
  font-style: normal;
  font-weight: 700;
  line-height: 15px;
  left: 0;
  min-height: 50px;
  position: sticky;
  padding: 10px;
  width: 130px;
`;

const SizeTable = () => {

  const arr = ['Heel-toe measurement','8.7"','8.9"','9.0"', '9.2"', '9.5"', '9.7"', '9.8"', '10.0"', '10.2"'];
  const arr2 = ['US - Mens','4','4.5','5', '5.5', '6', '6.5', '7', '7.5', '8'];
  const arr3 = ['US - Womens','4','4.5','5', '5.5', '6', '6.5', '7', '7.5', '8'];

  return (
    <Table>
      <TableInner>
        <TableRow>
        {arr.map(size =>
          <TableCell>
            <div key={size}>{size}</div>
          </TableCell>
        )}
        </TableRow>

        <TableRow>
        {arr2.map(size =>
          <TableCell2>
            <div key={size}>{size}</div>
          </TableCell2>
        )}
        </TableRow>

        <TableRow>
        {arr3.map(size =>
          <TableCell2>
            <div key={size}>{size}</div>
          </TableCell2>
        )}
        </TableRow>

      </TableInner>
    </Table>
  )
}




export default SizeTable;