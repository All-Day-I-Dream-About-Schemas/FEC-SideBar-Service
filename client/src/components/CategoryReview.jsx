import React from 'react';
import styled from 'styled-components';
import Stars from './Stars.jsx';
import Title from './Title.jsx';
import Price from './Price.jsx';
import LearnMore from './LearnMore.jsx';

const Wrapper = styled.div`
  display: flex;
  flex-flow: row norwrap;
  justify-content: space-between;
  text-align: center;
`;

const Category = styled.div`
  // flex: 1;
  font-family: AdihausDIN,Helvetica,Arial,sans-serif;
  font-weight: 400;
  text-transform: uppercase;
  letter-spacing: 2px;
`;

const Review = styled.div`
  display: flex;
  align-item: center;
  // flex: 1;
  // flex-flow: row norwrap;
  font-family: AdihausDIN,Helvetica,Arial,sans-serif;
  font-weight: 700;
  // position: center;
  // text-transform: uppercase;
  letter-spacing: 2px;

`;


const CategoryReview = () => (
  <div>
    <Wrapper>
      <Category>
        women . original
      </Category>
      <Review>
        <Stars />
        <Stars />
        <Stars />
        <Stars />
        <Stars />
        7753
      </Review>
    </Wrapper>
    <br></br>
    <Title />
    <br></br>
    <Price />
    <br></br>
    <LearnMore />
  </div>
);


export default CategoryReview;