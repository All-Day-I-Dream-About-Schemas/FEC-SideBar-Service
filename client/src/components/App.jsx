import React from 'react';
import CategoryReview from './CategoryReview.jsx';
import styled from 'styled-components';


const Body = styled.div`
  // display: flex;
  background-color: #F4EBF4;
  padding: 20px;
`;

class App extends React.Component {
  constructor() {
    super();
  }

  render() {
    return (
      <Body>
        <CategoryReview />
      </Body>
    )
  }
};


export default App;