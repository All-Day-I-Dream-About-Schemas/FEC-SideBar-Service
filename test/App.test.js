import React from 'react';
import Enzyme, { shallow, render, mount } from 'enzyme';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import sinon from 'sinon';


// import { expect } from 'chai';
import Adapter from 'enzyme-adapter-react-16';
import App from '../client/src/components/App';
import CategoryReview from '../client/src/components/CategoryReview';
import Discount from '../client/src/components/Discount'
import LearnMore from '../client/src/components/LearnMore';
import SizeTable from '../client/src/components/SizeTable';
import LearnMoreModal from '../client/src/components/LearnMoreModal'


Enzyme.configure({ adapter: new Adapter() });


describe('App',  () => {

  const wrapper = shallow(<App />);

  it('should render correctly in "debug" mode', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders one <LearnMore /> components', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find(LearnMore)).toHaveLength(1);
  });

  it('check for children', () => {
    expect(wrapper.children().length).toEqual(2)
  })
});

describe('Categoryreview', () => {
  it('should render correctley with no props', () => {
    const component = shallow(<CategoryReview />);
    expect(component).toMatchSnapshot();
  })
})


describe('LearnMore', () => {
  const wrapper = shallow(<LearnMore />);

  it('Learn component should render correctley with no props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('check for children', () => {
   expect(wrapper.children().length).toEqual(2)
 })

  it('should check for button existance', () => {
    const learnButton = wrapper.find('LearnMoreButton');
    expect(learnButton.exists()).toBe(true);
  });


  it('simulates click events', () => {
    const onButtonClick = sinon.spy();
    const wrapper = shallow((
      <LearnMore onButtonClick={onButtonClick} />
    ));
    wrapper.find('button').simulate('click');
    expect(onButtonClick).to.have.property('display', true);
  });


})
 describe('SizeTable should render properly', () => {
   it('renders list', () => {
     const wrapper = renderer.create(<SizeTable />)
     const arr = [5,6,7,9];
     expect(wrapper).toMatchSnapshot()
   })

 })

 describe('Discount', () => {
   const wrapper = shallow(<Discount />);
   const header = wrapper.find('styled.h5')

   it('Shoud check for H5 header', () => {
     expect(header).toBeDefined();
   })
 })




// describe('App', () => {
//   const wrapper = shallow(<App />);
//   it('should show the text', () => {
//     const text = wrapper.find('div');
//     expect(text.text()).toBe('WOMEN');
//   });
// });

// it('should render List with the correct styles', () => {
//   // const component = render(<LearnMore />)
//   const button = renderer.create(<LearnMore />).toJSON()

//   expect(button).toHaveStyleRule('cursor', 'pointer')
// })