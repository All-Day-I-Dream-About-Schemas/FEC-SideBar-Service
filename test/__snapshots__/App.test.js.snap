// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`Categoryreview should render correctley with no props 1`] = `ShallowWrapper {}`;

exports[`LearnMore Learn component should render correctley with no props 1`] = `ShallowWrapper {}`;

exports[`LearnMore should render correctley with no props 1`] = `ShallowWrapper {}`;

exports[`SizeTable should render properly renders list 1`] = `
<div
  className=""
>
  <div
    className=""
  >
    <div
      className=""
    >
      <div
        className=""
      >
        <div>
          Heel-toe measurement
        </div>
      </div>
      <div
        className=""
      >
        <div>
          8.7"
        </div>
      </div>
      <div
        className=""
      >
        <div>
          8.9"
        </div>
      </div>
      <div
        className=""
      >
        <div>
          9.0"
        </div>
      </div>
      <div
        className=""
      >
        <div>
          9.2"
        </div>
      </div>
      <div
        className=""
      >
        <div>
          9.5"
        </div>
      </div>
      <div
        className=""
      >
        <div>
          9.7"
        </div>
      </div>
      <div
        className=""
      >
        <div>
          9.8"
        </div>
      </div>
      <div
        className=""
      >
        <div>
          10.0"
        </div>
      </div>
      <div
        className=""
      >
        <div>
          10.2"
        </div>
      </div>
    </div>
    <div
      className=""
    >
      <div
        className=""
      >
        <div>
          US - Mens
        </div>
      </div>
      <div
        className=""
      >
        <div>
          4
        </div>
      </div>
      <div
        className=""
      >
        <div>
          4.5
        </div>
      </div>
      <div
        className=""
      >
        <div>
          5
        </div>
      </div>
      <div
        className=""
      >
        <div>
          5.5
        </div>
      </div>
      <div
        className=""
      >
        <div>
          6
        </div>
      </div>
      <div
        className=""
      >
        <div>
          6.5
        </div>
      </div>
      <div
        className=""
      >
        <div>
          7
        </div>
      </div>
      <div
        className=""
      >
        <div>
          7.5
        </div>
      </div>
      <div
        className=""
      >
        <div>
          8
        </div>
      </div>
    </div>
    <div
      className=""
    >
      <div
        className=""
      >
        <div>
          US - Womens
        </div>
      </div>
      <div
        className=""
      >
        <div>
          4
        </div>
      </div>
      <div
        className=""
      >
        <div>
          4.5
        </div>
      </div>
      <div
        className=""
      >
        <div>
          5
        </div>
      </div>
      <div
        className=""
      >
        <div>
          5.5
        </div>
      </div>
      <div
        className=""
      >
        <div>
          6
        </div>
      </div>
      <div
        className=""
      >
        <div>
          6.5
        </div>
      </div>
      <div
        className=""
      >
        <div>
          7
        </div>
      </div>
      <div
        className=""
      >
        <div>
          7.5
        </div>
      </div>
      <div
        className=""
      >
        <div>
          8
        </div>
      </div>
    </div>
  </div>
</div>
`;
